Create a separate environment for testing schematic refactor branch

linglp opened the pull request:
PR Checklist:
[ ] Describe and explain your intentions for this change
[ ] Setup pre-commit and run the validators (info in README.md)

Related to: https://sagebionetworks.jira.com/browse/FDS-1233

We need to create a separate schematic environment for testing schematic refactor. Ideally, the environment should be like "dev" environment that is under D&T dev account

linglp committed:
add dev refactor env

linglp committed:
make sure stack name prefix is unique

linglp committed:
remove empty

linglp committed:
pre-commit fix

zaro0508 commented on .github/workflows/aws-deploy.yml:
```diff
@@ -90,6 +90,37 @@ jobs:
+
+  cdk-deploy-dev-refactor:
+    runs-on: ubuntu-latest
+    if: ${{ github.ref == 'refs/heads/dev' }}
```
shouldn't this execute based on a different branch?

linglp replied:
Yes. Thank you for the comment @zaro0508 !

zaro0508 commented on .github/workflows/aws-deploy.yml:
```diff
@@ -90,6 +90,37 @@ jobs:
+        uses: youyo/aws-cdk-github-actions@v2
+        with:
+          cdk_subcommand: 'deploy'
+          cdk_args: '--require-approval never --context env=dev --progress events --debug'
```
`env` should probably reference `dev-refactor`?

linglp replied:
Yes. Good catch. Thank you

zaro0508 reviewed (changes requested):
i believe [secrets need to be setup for each environment](https://github.com/Sage-Bionetworks/schematic-infra#secrets), have you setup secrets yet?

linglp committed:
update env name

linglp commented:
> i believe [secrets need to be setup for each environment](https://github.com/Sage-Bionetworks/schematic-infra#secrets), have you setup secrets yet?

Can I name my secret as: `schematic-dev-refactor-DockerFargateStack/dev-refactor/ecs`. Based on the documentation, the naming convention is: `<cfn stack id>/<environment id>/<secret name>`. Since my stack name prefix is: `schematic-dev-refactor`, so I assume that I should use `schematic-dev-refactor`.. but based on the screen shot of the failed deployment, it seems like the stack name is: `schematic/dev-refactor-dev-refactor-DockerFargateStack`? 
<img width="794" alt="Screen Shot 2023-11-03 at 9 47 17 AM" src="https://github.com/Sage-Bionetworks/schematic-infra/assets/55448354/f7ece76d-e496-49f2-b28d-4f3191eb1e17">

What is the environment id? Should it just be `dev` or `dev-refactor`?

zaro0508 commented on cdk.json:
```diff
@@ -41,6 +41,17 @@
+      },
+      "STACK_NAME_PREFIX": "schematic-dev-refactor",
+      "ACM_CERT_ARN": "arn:aws:acm:us-east-1:631692904429:certificate/0e9682f6-3ffa-46fb-9671-b6349f5164d6",
+      "VPC_CIDR": "10.255.73.0/24"
```
this CIDR is already used by `dev` env.  i suggest you pick a CIDR value that has not been used yet.  maybe "10.255.76.0/24"? more info about it here https://sagebionetworks.jira.com/wiki/spaces/IT/pages/2850586648/Setup+AWS+VPC

linglp replied:
Okay thanks for pointing this out. I will update this part.

zaro0508 commented:
please take a look at the code that defines the <cfn stack id> , https://github.com/Sage-Bionetworks/schematic-infra/blob/dev/docker_fargate/docker_fargate_stack.py#L47

it's a bit of a concatenation of the stack prefix, the context and `DockerFargateStack`

linglp committed:
update vpc cidr value

linglp committed:
update vpc cidr

linglp committed:
Delete latency-box-plot.svg

linglp committed:
update vpc cidr value back

linglp committed:
Merge branch 'dev-refactor' of https://github.com/Sage-Bionetworks/schematic-infra into dev-refactor

linglp commented:
> please take a look at the code that defines the , https://github.com/Sage-Bionetworks/schematic-infra/blob/dev/docker_fargate/docker_fargate_stack.py#L47
> 
> it's a bit of a concatenation of the stack prefix, the context and `DockerFargateStack`

Thanks Khai. I think maybe the secret is set up correctly? I could see that the secret got retrieved on Nov.5th..

<img width="1103" alt="Screen Shot 2023-11-06 at 3 40 53 PM" src="https://github.com/Sage-Bionetworks/schematic-infra/assets/55448354/10fdc043-5427-4113-9ce6-3d16557c3794">

linglp commented:
@zaro0508 I noticed that the github action failed because the CIDR value was still referencing the wrong one after I updated it. After I deleted the old stack and redeployed again, it seems to work. 


Old stack: 
<img width="866" alt="Screen Shot 2023-11-06 at 4 03 08 PM" src="https://github.com/Sage-Bionetworks/schematic-infra/assets/55448354/b77419e5-0a40-4d91-b7c2-7bf294d91107">

New stack: 

<img width="580" alt="Screen Shot 2023-11-06 at 4 06 22 PM" src="https://github.com/Sage-Bionetworks/schematic-infra/assets/55448354/3bd738d8-9f06-47cf-86de-035c5f7fde1b">
